client/src/pages/ExploreCourses.jsx

```diff
@@ -1,91 +1,137 @@
+        }
+    }, [accountOperationsResponse, setIsModalShown]);
+
+    const { enrollCourse } = accountOperationsMethods;
+    const enrollHandler = () => {
         const request = new EnrollCourseRequest();
-        request.setCourseid(courseId);
+        request.setCourseid(selectedCourse.getId());
         enrollCourse(request, user.token);
     };
 
+    let modalContent = (
+        <>
+            Are you sure you want to proceed buying{' '}
+            <span style={{ fontWeight: 'bold' }}>
+                {selectedCourse && selectedCourse.getTitle()}
+            </span>
+            ?
+        </>
+    );
+    if (accountOperationsState.error) {
+        modalContent = 'Error occurred';
+    }
+
     return (
-        <Container style={{ maxWidth: '80rem', margin: '2rem auto' }}>
-            <CourseList
-                isLoading={false}
-                error={false}
-                header="My Courses"
-                missingCoursesMessage="You don't have any courses"
+        <>
+            <Modal
+                open={isModalShown}
+                onClose={() => setIsModalShown(false)}
+                size="tiny"
+                centered={false}
+                closeIcon
             >
-                {courses.map(course => (
-                    <Item key={course.getId()} as="li">
-                        <Item.Image size="tiny" src={courseImage} />
-                        <Item.Content>
-                            <Item.Header content={course.getTitle()} />
-                            <Item.Meta
-                                content={course.getTeacher().getName()}
-                            />
-                            <Item.Description
-                                content={course.getDescription()}
-                            />
-                            {user ? (
-                                <Item.Extra>
-                                    <Button
-                                        onClick={() =>
-                                            enrollHandler(course.getId())
-                                        }
-                                        icon="cart"
-                                        floated="right"
-                                        color="blue"
-                                    />
-                                </Item.Extra>
-                            ) : null}
-                        </Item.Content>
-                    </Item>
-                ))}
-            </CourseList>
-        </Container>
+                <ConfirmationModalBody
+                    title="Buy a course"
+                    isLoading={accountOperationsState.isLoading}
+                    onClose={() => setIsModalShown(false)}
+                    onConfirm={enrollHandler}
+                >
+                    {modalContent}
+                </ConfirmationModalBody>
+            </Modal>
+            <Container style={{ maxWidth: '80rem', margin: '2rem auto' }}>
```

Comment:
is it better way than each attribute on a new line?

Reply:
The line is under 80 chars so I prefer to keep it compact